### PR TITLE
fix(security): upgrade dependencies and resolve bundler-audit vulnerabilities

### LINKED ISSUE #7309: 🐞 Bug: bundler-audit failing due to multiple vulnerable dependencies
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Running `bundle exec bundler-audit --update` reports multiple known vulnerabilities
in dependencies currently present in Gemfile.lock.

These vulnerabilities are causing CI to fail even on the latest master branch,
and are not related to recent changes in test code.

Affected gems include:
- activestorage (CVE-2026-33658)
- avo (CVE-2026-33209)
- bcrypt (CVE-2026-33306)
- devise (CVE-2026-32700)
- json (CVE-2026-33210)

### Steps To Reproduce

1. Run `bundle exec bundler-audit --update`
2. Observe reported vulnerabilities and non-zero exit status

### Expected Behavior

`bundler-audit` should pass without reporting known vulnerabilities,
and CI should not fail due to dependency security issues.

### Screenshots

_No response_

### Device Information [optional]

```markdown
- OS:
- Browser:
- version:
```

### Additional context

_No response_

### Are you working on this issue?

Yes

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #7309

Depends on PR #7308 

#### Describe the changes you have made in this PR -

This PR upgrades dependencies to resolve vulnerabilities reported by `bundler-audit`.

### Updated dependencies:
- `avo` → >= 3.30.3 (fixes CVE-2026-33209)
- `bcrypt` → >= 3.1.22
- `json` → >= 2.19.2
- `devise` → >= 5.0.3
- `rails` → updated to include ActiveStorage fix for CVE-2026-33658

### Test updates:
- Updated Devise-related system tests to match new default authentication message:
  - `"Invalid Email or password."` → `"Invalid email or password."`

### Verification:
- All RSpec tests pass successfully
- `bundle exec bundler-audit --update` reports no vulnerabilities

---

### Screenshots of the UI changes (If any) -
N/A (dependency and test updates only)

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

This PR resolves security vulnerabilities by upgrading affected dependencies while ensuring application stability.

Each dependency was upgraded incrementally and verified by running the full test suite. Special care was taken with Devise and Rails upgrades, as they introduce behavior changes.

Devise 5 changes default authentication messages and enforces stricter confirmable behavior. Corresponding system tests were updated to reflect these changes.

The Avo upgrade was previously reverted, so additional verification was performed to ensure compatibility with the current codebase.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved sign-in error message capitalization for better consistency and clarity.

### CODE REVIEW

File: spec/system/signin_spec.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
     # Percy snapshot here to see the error state
     page.percy_snapshot("Sign In - No Credentials")
 
-    expect(page).to have_text("Invalid Email or password.")
+    expect(page).to have_text("Invalid email or password")
```
Comment (naman79820) [inline spec/system/signin_spec.rb:21]: Updated signin system tests to reflect Devise 5 default authentication message.

Devise >= 5.0.3 introduces a change in the default error message
("Invalid Email or password." → "Invalid email or password.").
Tests have been updated to match this behavior.

**Comment by naman79820** [comment]
All checks passed :))
